scripts/core/kernel.yml

```diff
@@ -128,18 +128,20 @@ params:
     - type: $x_kernel_info_t
       name: propName
       desc: "[in] name of the Kernel property to query"
-    - type: "size_t*"
+    - type: "size_t"
       name: propSize
       desc: |
-            [in,out] pointer to the size of the Kernel property value
-            If *propSize is 0 or greater than the number of bytes of the Kernel property,
-            the call shall update the value with actual number of bytes of the Kernel property.            
+            [in] the size of the Kernel property value.           
     - type: "void*"
       name: propValue
       desc: |
-            [in,out][optional][range(0, *propSize)] value of the Kernel property.
-            If *propSize is less than the number of bytes for the Kernel property,
-            only the first *propSize bytes will be returned.
+            [in,out][optional][range(0, propSize)] value of the Kernel property.
+            If propSize is less than the number of bytes for the Kernel property,
```

Comment:
Should we align on naming here? In some places input size parameters are named `Size` but then in entry points like this one we have `propSize`.

Reply:
This behaviour is also slightly different to other entry points which will fail if there isn't enough memory in `propValue` to hold the queried property.

Reply:
I think I prefer `propSize` over just `Size`. I'll update if no objections.

As you your second point. I think I'll update this to be like openCL, fail if `!size>=paramValue`